apps/docs/src/components/landing/FeatureSection.tsx

```diff
@@ -0,0 +1,90 @@
+import { faArrowRight } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { useState, useEffect } from 'react';
+interface Feature {
+  title: string;
+  icon: any; // You can specify the correct type for FontAwesomeIcon
+  text: string;
+  hoverImage: string;
+  shadow?: boolean;
+}
+
+interface FeatureSectionProps {
+  title: string;
+  description: string;
+  features: Feature[];
+}
+
+function FeatureSection({ title, description, features }: FeatureSectionProps) { 
+  const [hoveredImage, setHoveredImage] = useState(features[0]?.hoverImage || '');
+  const [hoveredFeature, setHoveredFeature] = useState<Feature | null>(features[0]); // Set the first feature as default
+
+  useEffect(() => {
+    if (features.length > 0) {
+      setHoveredImage(features[0].hoverImage);
+      setHoveredFeature(features[0]); // Initialize with the first feature
+    }
+  }, [features]);
+
+  const handleMouseEnter = (feature: Feature) => {
+    setHoveredImage(feature.hoverImage);
+    setHoveredFeature(feature);
+  };
+
+  const handleMouseLeave = () => {
+    setHoveredFeature(features[0]); // Reset to first feature when mouse leaves
+    setHoveredImage(features[0].hoverImage); // Reset image to first feature's image
+  };
+
+  return (
+    <div className="overflow-hidden bg-white py-16 sm:py-24">
+      <div className="mx-auto max-w-7xl px-6 lg:px-8">
+        <div className="mx-auto grid max-w-2xl grid-cols-1 gap-x-8 gap-y-16 sm:gap-y-20 lg:mx-0 lg:max-w-none lg:grid-cols-2">
+          <div className="lg:pr-8 lg:pt-4">
+            <div className="lg:max-w-lg">
+              <h2 className="text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl">
+                {title}
+              </h2>
+              <p className="mt-6 text-lg leading-8 text-gray-600">
+                {description}
+              </p>
+              <dl className=" mt-10 max-w-xl space-y-8 text-base leading-7  text-gray-600 lg:max-w-none">
```

Comment:
Instead of spacing between the elements, would it maybe make sense to use padding on the components themselves? In this way the UI does not switch back to the first element when one changes from the second to the third?